lib/compat/wordpress-6.7/blocks.php

```diff
@@ -43,3 +43,33 @@ function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_allow_variations_php_file', 10, 2 );
+
+/**
+ * Filter the query vars (tax_query) for the Query Loop block to support post formats.
+ *
+ * @param array    $query The query vars.
+ * @param WP_Block $block Block instance.
+ * @return array   The filtered query vars.
+ */
+function gutenberg_filter_query_loop_block_query_vars_post_format( $query, $block ) {
+	if ( isset( $block->context['query']['format'] ) &&
+		! empty( $block->context['query']['format'] ) ) {
+		// Add the required "post-format-" prefix to each format.
+		$formats = array_map(
+			function ( $format ) {
```

Comment:
```suggestion
			static function ( $format ) {
```

I think even closures outside classes should be `static`.